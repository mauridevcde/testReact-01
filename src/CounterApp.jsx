import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  const handledAdd = () => {
    setCount(count + 1);
  };
  const handledRestar = () => {
    setCount(count - 1);
  };
  const handledReset = () => {
    setCount(value);
  };

  return (
    <div>
      <h1>CounterApp</h1>
      {<h2>{count}</h2>}
      <button onClick={handledAdd}>+1</button>
      <button onClick={handledRestar}>-1</button>
      <button onClick={handledReset}>Reset</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
