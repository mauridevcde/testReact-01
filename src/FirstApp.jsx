import PropTypes from "prop-types"

export const FirstApp = ({title, name}) => {
    return (
        <div>
            <h1>{title}, {name}</h1>
        </div>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: "esta es una props default",
    name: "mauricio"
};