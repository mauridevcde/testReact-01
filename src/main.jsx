import React from 'react'
import ReactDOM from 'react-dom/client'
import {CounterApp} from './CounterApp'
import {FirstApp} from './FirstApp'
import './app.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
   <FirstApp  title="Hola soy mauricio" />
  </React.StrictMode>
) 
