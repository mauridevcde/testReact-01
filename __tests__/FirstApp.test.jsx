import { render, screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <First app/>", () => {
//   test("Debe hacer match con el snapshot", () => {
//     const title = "Hola mundo";

//     const { container } = render(<FirstApp title={title} />);
//     //console.log(container);
//     expect(container).toMatchSnapshot(); //snapshot es una imagen de un componente, comprueba si el componente es igual a la imagen
//   })


  test("Debe mostrar el titulo en un H1", () => {

    const title = "Hola, soy Goku"
    const { container  } = render(<FirstApp title={title}/>)
    expect (screen.getByText(title)).toBeTruthy()
  })
  
})
