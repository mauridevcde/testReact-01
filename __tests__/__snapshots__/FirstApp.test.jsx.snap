// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <First app/> Debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      Hola mundo
      , 
      mauricio
    </h1>
  </div>
</div>
`;
